tools/src/main/java/org/apache/kafka/trogdor/workload/ConnectionStressWorker.java

```diff
@@ -103,29 +103,17 @@ public void start(Platform platform, WorkerStatusTracker status,
         synchronized (ConnectionStressWorker.this) {
             this.totalConnections = 0;
             this.totalFailedConnections = 0;
-            this.nextReportTime = 0;
             this.startTimeMs = TIME.milliseconds();
         }
-        this.throttle = new ConnectStressThrottle(WorkerUtils.
-            perSecToPerPeriod(spec.targetConnectionsPerSec(), THROTTLE_PERIOD_MS));
-        this.workerExecutor = Executors.newFixedThreadPool(spec.numThreads(),
+        this.workerExecutor = Executors.newScheduledThreadPool(1 + spec.numThreads(),
```

Comment:
IMO, this would be a bit cleaner with a separate executor with a single thread for the status updater.

Reply:
Good idea.  I put this in a separate executor.